hack/build-images.sh

```diff
@@ -1,14 +1,75 @@
 #!/bin/bash
 
-set -ex
+set -o errexit
+set -o nounset
+set -o pipefail
 
-prefix=${PREFIX:-${1:-openshift/origin-}}
-version=${VERSION:-${2:-latest}}
-for component in fluentd elasticsearch kibana deployment curator ; do
-    docker build -t "${prefix}logging-${component}:${version}"       ../$component/
+STARTTIME=$(date +%s)
+
+prefix="${PREFIX:-openshift/origin-}"
+version="${VERSION:-latest}"
+os_tag="${OS_PUSH_TAG:-latest}"
+
+source_root=$(dirname "${0}")/..
+
+# update the FROM statement in the Dockerfile to be a specific
+# tag.  Assumes there is only 1 FROM but will change them all
+# if there is more than one
+function create_dockerfile_using_base_tag {
```

Comment:
1. I have no idea how far away we would be from that because there are all the associated scripts that get built into the deployer image that need to be validated.  Additionally, @richm has been working on CI for logging that executes e2e tests which need to be validated.  We need to think through what has to happen and put it through some testing cycles.  I imagine its at least a sprint worth of effort.
2. This allows us to have a working Dockerfile and additionally be able to modify the base image as needed.  Is that not desirable?